x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/helpers.ts

```diff
@@ -77,16 +77,10 @@ export function getQueryLanguageLabel(language: string) {
 }
 
 /**
- * Assigns type `Filter` to items that have a `meta` property. Removes any other items.
+ * Assigns type `Filter[]` to an array if every item in has a `meta` property.
  */
-export function typeCheckFilters(filters: unknown[]): Filter[] {
-  return filters.filter((f) => {
-    if (typeof f === 'object' && f !== null && 'meta' in f) {
-      return true;
-    }
-
-    return false;
-  }) as Filter[];
+export function isFilters(maybeFilters: unknown[]): maybeFilters is Filter[] {
+  return maybeFilters.every((f) => typeof f === 'object' && f !== null && 'meta' in f);
```

Comment:
Should `meta`'s type be checked as well?

Reply:
Thanks! Added in https://github.com/elastic/kibana/pull/192342/commits/f89e47613bf528de50571615fbe2a0e6b8c53d7a.